### DISCUSSION TITLE
How to know default cmd line parameters for clangd

### DISCUSSION THREAD

**Discussion by f-squirrel** [clangd/clangd#1389] [Q&A]
Hi folks,

I am using `clangd` via VS Code extension. I am trying to understand what flags to add or what flags are on by default.
For example, shall I add the following flags to the cmd line:

* `--clang-tidy`
* `--function-arg-placeholders`
* `--header-insertion=iwyu`

**Comment by HighCommander4** [comment]
The available command line options can be listed with `clangd --help`.

I'm not aware of a place where the default values are documented besides the [source code](https://searchfox.org/llvm/rev/8db98599f16eb9894a2047926cf9090905521197/clang-tools-extra/clangd/tool/ClangdMain.cpp#278,283) (and [sometimes](https://searchfox.org/llvm/rev/8db98599f16eb9894a2047926cf9090905521197/clang-tools-extra/clangd/tool/ClangdMain.cpp#244,249) tracking down the default values requires some [digging](https://searchfox.org/llvm/rev/8db98599f16eb9894a2047926cf9090905521197/clang-tools-extra/clangd/CodeComplete.h#97)).

It might be a good idea to document the default values on the [clangd website](https://clangd.llvm.org/), though we'd have to be careful to keep that documentation in sync with code changes.

**Comment by tom-anders** [comment]
Why not just let --help print the default values though? Would this require non-trivial changes to the command line parser code?

**Reply by HighCommander4** [reply]
It looks like this would need to be implemented upstream in libLLVMSupport, but I do agree this would be nicer than documenting the default values in a separate place that can potentially get out of sync.
